src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -354,6 +354,17 @@ static int add_swap(const char *path) {
                 return 0;
         }
 
+        const char *path;
+        _cleanup_free_ char *crypto_what = NULL;
+        if (streq_ptr(p->fstype, "crypto_LUKS")) {
+                r = add_cryptsetup("swap", p->node, true, true, &crypto_what);
```

Comment:
Can there be multiple swap partitions? If so, this shouldn't be hard-coded

Reply:
True. I'll do something like `"swap" + p->uuid`.

Reply:
I think this should be something less likely to conflict and more self-explanatory. Maybe "gpt-auto-swap-<uuid>".

Reply:
Actually the discoverable partitions spec documents the DM name to use for the partitions. It should be augmented to also do so for the swap device, and I thinkg the name /dev/mapper/swap is actually fine. I mean, we bypass all this anyway if there's any form of swap explicitly configured, hence I think the simple, obvious name is good, because it's not going to result in conflicts but be very very self documenting.

Reply:
Indeed, DISCOVERABLE_PARTITIONS has various names like this: `/dev/mapper/root`, `/dev/mapper/tmp`, etc. Using `/dev/mapper/swap` is in line with this.

Nevertheless, I think the general approach is problematic. There is no conflict between various gpt-auto names as defined by us, but this namespace is not only used for this purpose. `/dev/mapper/` is shared for all device-mapper volumes, so not using any namespacing here is problematic. So I think all the names defined in DISCOVERABLE_PARTITIONS should be changed to `gpt-auto-*`. This would also make the whole scheme more obvious when used.

Reply:
I am not sure I agree, I'd much rather stick to our defensive approach: use the obvious names, but whenever we see any sign that the user uses this otherwise step aside. i.e. if we find /dev/mapper/swap in fstab then we should step away.

I see nothing wrong with using the obvious names as long as we make it highly unlikely to cause a clash with this.

Renaming the devices is problematic since they are kinda API. If we change them this needs a really really strong reason, and I really don't see that here...

Reply:
Yeah, renaming at this point would be complicated.

> we make it highly unlikely to cause a clash with this.

We are using some very short and "obvious" names in a shared namespace. "swap" is not so bad, but "tmp" is something that is very likely to be used for a different purpose by an admin.

Reply:
What would be the issues with changing the mapper names? They are used internally by the generated mount units, no? As far as I can see it's the mount points that need to stay stable, no?